Project Walking/Source code/app/src/main/res/layout/activity_main.xml

```diff
@@ -6,13 +6,21 @@
     android:layout_height="match_parent"
     tools:context=".MainActivity">
 
+    <org.osmdroid.views.MapView
+        android:id="@+id/map"
+        android:layout_width="fill_parent"
+        android:layout_height="fill_parent"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent" />
+
     <TextView
+        android:id="@+id/helloWorld"
         android:layout_width="wrap_content"
         android:layout_height="wrap_content"
         android:text="Hello World!"
-        app:layout_constraintBottom_toBottomOf="parent"
-        app:layout_constraintLeft_toLeftOf="parent"
-        app:layout_constraintRight_toRightOf="parent"
-        app:layout_constraintTop_toTopOf="parent" />
+        app:layout_constraintStart_toStartOf="@+id/map"
+        app:layout_constraintTop_toTopOf="@+id/map" />
+
```

Comment:
The element `<com.google.android.material.floatingactionbutton.FloatingActionButton ... />` is missing here, so right now there is no button for the camera in your branch. The button exists in the master branch, and the rest of the camera code is in your branch, so the button must have disappeared when you merged the code. I suggest that you revert your last commit, start over with the changes from the previous review, and add them one by one. 

One commit for fixing activity_main.xml. 
One commit for adding the `MapController` class. 
One commit for using the `MapController` class in `MainActivity`. 
One commit for merging the master branch with your branch. 

I don't want to come across as unpleasant or rude in any way, but if the commits were done in that way before, you would only have to revert the "merge commit", but you could have kept your other changes. Now, since everything was made in one commit, you will have to start over and take care of the comments from the first review again, plus this review 😅